### DISCUSSION TITLE
A toggle state function

### DISCUSSION THREAD

**Discussion by jonbng** [GrandMoff100/HomeAssistantAPI#50] [Ideas]
Hey, this module is amazing! What about a toggle state function? Sorry if this is dumb or if its allready there. I just think it could be usefull. If you dont have time or dont want to, thats totally ok. Thanks anyway, i will be using this module a lot, i think. Sorry for bad English :)

**Comment by jonbng** [comment]
Just realized that this is probably dumb. I didnt understand the modules fully (And still dont) Sorry

**Comment by GrandMoff100** [comment]
No worries! A toggle state function could be useful for manual state setting. (Toggling states that aren't connected to real life entities. Entities only stored in Homeassistant.) But a toggle state function would only apply to Binary Sensors and it's implementations (like switch, or cover.) But because hard coding entity type specific methods isn't possible based on the structure of this library due to the dynamic nature of entity types.

So we probably won't implement that. But cool idea though.

**Reply by GrandMoff100** [reply]
HomeassistantAPI is probably overdue for some more love in the coming weeks though... Stay tuned!

**Reply by jonbng** [reply]
Oh thats nice! Enjoying the module so far :)
